cfgen/cfgen

```diff
@@ -1261,6 +1270,31 @@ class ConfPool(ToDhall):
                           f'pvers = {oids_to_dhall(self.pvers)}'])
         return '{ %s }' % args
 
+    @classmethod
+    def gen_sns_tolerance(cls, m0conf: Dict[Oid, Any],
+                          pool_desc: Dict[str, Any]) -> Failures:
+        encls_nr = len(cluster_encls(m0conf))
+        ctrls_per_encl = floor(len(cluster_ctrls(m0conf)) / encls_nr)
+        data_units = pool_desc['data_units']
+        parity_units = pool_desc['parity_units']
+        spare_units = pool_desc.get('spare_units', 0)
+        total_units = data_units + parity_units + spare_units
```

Comment:
While calculating the pool width, we need to consider this formula here? : 
https://jts.seagate.com/browse/EOS-23178?focusedCommentId=2164896&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2164896

Reply:
Yeah, so for now we discard asymmetric configurations which do not satisfy the symmetric pool width >= N + K + S.

Reply:
But this is a separate pool/pver validation issue.